### DISCUSSION TITLE
Document about Ceph for storage

### DISCUSSION THREAD

**Discussion by gobiursus** [apache/cloudstack#9862] [Q&A] [answered]
##### ISSUE TYPE

 * Documentation Report

##### COMPONENT NAME

~~~
Documentation and Ceph as storage
~~~

##### CLOUDSTACK VERSION

~~~
All
~~~

##### OS / ENVIRONMENT

Debian 12.7 Stable

##### SUMMARY

Can we add Ceph storage support into documentation? I want to use Ceph RBD as storage.

**Comment by kiranchavala** [comment, marked as answer]
@gobiursus the ceph storage documentation is mentioned here 

https://docs.cloudstack.apache.org/en/4.19.1.1/installguide/configuration.html#adding-primary-storage

![Screenshot 2024-10-17 at 4 48 56 PM](https://github.com/user-attachments/assets/1953dc01-7545-456e-b092-e21eaeb49cab)

**Comment by gobiursus** [comment]
Thank you. Is there anything I need to notice before I create RBD on Ceph? Like "I need to create 2 pools one for primary storage the other one for secondary storage" or "I need to create multiple pools one for this service and one for the other service like Openstack"?

**Comment by weizhouapache** [comment]
> Thank you. Is there anything I need to notice before I create RBD on Ceph? Like "I need to create 2 pools one for primary storage the other one for secondary storage" or "I need to create multiple pools one for this service and one for the other service like Openstack"?

It is totally up to you.
You just need to create a pool for clodstack. For the rest space, you can do anything you want.

**Comment by gobiursus** [comment]
> > Thank you. Is there anything I need to notice before I create RBD on Ceph? Like "I need to create 2 pools one for primary storage the other one for secondary storage" or "I need to create multiple pools one for this service and one for the other service like Openstack"?
> 
> It is totally up to you. You just need to create a pool for clodstack. For the rest space, you can do anything you want.

Thank you. Is secondary storage necessary? Is secondary storage just for a backup in Cloudstack?

**Comment by weizhouapache** [comment]
> > > Thank you. Is there anything I need to notice before I create RBD on Ceph? Like "I need to create 2 pools one for primary storage the other one for secondary storage" or "I need to create multiple pools one for this service and one for the other service like Openstack"?
> > 
> > 
> > It is totally up to you. You just need to create a pool for clodstack. For the rest space, you can do anything you want.
> 
> Thank you. Is secondary storage necessary? Is secondary storage just for a backup in Cloudstack?

secondary storage is required. it stores templates and snapshots.
However, you can not use ceph RBD pool as a secondary storage. 
you can create cephfs , export it over NFS and add to cloudstack as a secondary storage.

**Comment by gobiursus** [comment]
> > > > Thank you. Is there anything I need to notice before I create RBD on Ceph? Like "I need to create 2 pools one for primary storage the other one for secondary storage" or "I need to create multiple pools one for this service and one for the other service like Openstack"?
> > > 
> > > 
> > > It is totally up to you. You just need to create a pool for clodstack. For the rest space, you can do anything you want.
> > 
> > 
> > Thank you. Is secondary storage necessary? Is secondary storage just for a backup in Cloudstack?
> 
> secondary storage is required. it stores templates and snapshots. However, you can not use ceph RBD pool as a secondary storage. you can create cephfs , export it over NFS and add to cloudstack as a secondary storage.

How much space do we usually give to the secondary storage?

**Comment by weizhouapache** [comment]
> > > > > Thank you. Is there anything I need to notice before I create RBD on Ceph? Like "I need to create 2 pools one for primary storage the other one for secondary storage" or "I need to create multiple pools one for this service and one for the other service like Openstack"?
> > > > 
> > > > 
> > > > It is totally up to you. You just need to create a pool for clodstack. For the rest space, you can do anything you want.
> > > 
> > > 
> > > Thank you. Is secondary storage necessary? Is secondary storage just for a backup in Cloudstack?
> > 
> > secondary storage is required. it stores templates and snapshots. However, you can not use ceph RBD pool as a secondary storage. you can create cephfs , export it over NFS and add to cloudstack as a secondary storage.
> 
> How much space do we usually give to the secondary storage?

It depends on your use case.
For testing env, 50/100GB is enough.
For production env, 1TB is probably not enough.
